x/collection/module/module.go

```diff
@@ -0,0 +1,180 @@
+package module
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	"github.com/grpc-ecosystem/grpc-gateway/runtime"
+	"github.com/spf13/cobra"
+
+	"cosmossdk.io/core/address"
+	"cosmossdk.io/core/appmodule"
+	"cosmossdk.io/core/store"
+	"cosmossdk.io/depinject"
+
+	"github.com/cosmos/cosmos-sdk/client"
+	"github.com/cosmos/cosmos-sdk/codec"
+	codectypes "github.com/cosmos/cosmos-sdk/codec/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/types/module"
+
+	modulev1 "github.com/Finschia/finschia-sdk/api/lbm/collection/module/v1"
+	"github.com/Finschia/finschia-sdk/x/collection"
+	"github.com/Finschia/finschia-sdk/x/collection/client/cli"
+	"github.com/Finschia/finschia-sdk/x/collection/keeper"
+)
+
+// ConsensusVersion defines the current x/collection module consensus version.
+const ConsensusVersion = 2
+
+var (
+	_ module.AppModuleBasic = AppModule{}
+	_ module.HasGenesis     = AppModule{}
+	_ module.HasServices    = AppModule{}
+
+	_ appmodule.AppModule = AppModule{}
+)
+
+// AppModuleBasic defines the basic application module used by the collection module.
+type AppModuleBasic struct {
+	cdc codec.Codec
```

Comment:
`codec.Codec` contains the address codec. After updating the signatures of `NewQueryCmd()` and `keeper.InitGenesis()`, the separate field would have no advantage.